content/PIV/Guides/SSH_with_PIV_and_PKCS11.adoc

```diff
@@ -1,76 +1,59 @@
 * a YubiKey with the PIV application loaded
-* the yubico-piv-tool software
+* the yubico-piv-tool software (download from link:../yubico-piv-tool/Releases/)
 * the OpenSC software
 * OpenSSH
-** If you are using OSX El Capitan (10.11) or earlier, for ssh-agent to work a newer OpenSSH than is delivered with the system; macOS Sierra (10.12) contains a compatible version
+* For OS X, iOS 10.13 and later are supported
 
 [NOTE]
-The following example assume that you have not yet changed the management key. If you have changed the management key, add `--key` to the `yubico-piv-tool -a import-certificate` command below.
+RSA 4096-bit keys are not currently supported due to a limitation in the PIV spec: https://github.com/Yubico/yubico-piv-tool/issues/58
+
+[NOTE]
+The examples given in the following steps assume that you have not yet changed the management key. If you have changed the management key, add `--key` to the `yubico-piv-tool -a import-certificate` command below.
 
 
-=== Steps
-1. Generate or import a key in slot 9a (any slot should suffice):
-a. Import the key (PEM format):
+=== Procedure
+*Step 1*: Import or generate a key in slot 9a (any slot should suffice):
+
+Either *import* the key (PEM format):
 
   $ yubico-piv-tool -s 9a -a import-key -i key.pem
-+
+
 [NOTE]
-If an external key is imported and there already exists a certificate step 2 can be skipped.
+If an external key has been imported and a certificate already exists, skip step 2.
 
-b. Generate the key:
+or *generate* the key:
 
   $ yubico-piv-tool -s 9a -a generate -o public.pem
-+
-[NOTE]
-RSA 4096-bit keys are not currently supported due to a limitation in the PIV spec: https://github.com/Yubico/yubico-piv-tool/issues/58
 
-2. Create a self-signed certificate for that key.
-The only use for the X.509 certificate is to make PIV/PKCS#11 lib happy.
-They would want to be able to extract the public-key from the smartcard,
-and do that through the X.509 certificate.
+*Step 2*: Create a self-signed certificate for that key. The only use for the X.509 certificate is to satisfy PIV/PKCS #11 lib. It needs to be able to extract the public-key from the smartcard, and to do that through the X.509 certificate.
 
   $ yubico-piv-tool -a verify-pin -a selfsign-certificate -s 9a -S "/CN=SSH key/" -i public.pem -o cert.pem
 
-3. Load the certificate:
+*Step 3*: Load the certificate:
 
    $ yubico-piv-tool -a import-certificate -s 9a -i cert.pem
 
-4. Find out where OpenSC has installed the pkcs11 module.
-
-  * For OS X with binary installation this is typically in `/Library/OpenSC/lib/`. Homebrew users can use:
-
-  $ sudo ln `brew list opensc | grep lib/opensc-pkcs11.so` /usr/local/lib/opensc-pkcs11.so
+*Step 4*: Find out where OpenSC has installed the pkcs #11 module. For OS X with binary installation, this is typically in `/Library/OpenSC/lib/`. For a Debian-based system, this is typically in `/usr/lib/x86_64-linux-gnu/`. From now on, we will call this location `$OPENSC_LIBS`.
```

Comment:
Now that we have a functional pkcs11 module (the ykcs11), we can use that instead of OpenSC's. For my debian based system, the ykcs11 module ends up in /usr/local/lib/libykcs11.so (/usr/lib/libykcs11.so is also a common location). On my colleagues macos, it is in /usr/local/lib/libykcs11.dylib

Reply:
Done. And I removed the OpenSC as a prerequisite, too, adding why it is no longer needed.